tests/ecs/ambiguity_detection.rs

```diff
@@ -0,0 +1,120 @@
+//! An example to confirm that `bevy` doesn't have system order ambiguity with [`DefaultPlugins`]
+//! This is run in CI to ensure that this doesn't regress again.
+
+use bevy::{
+    ecs::schedule::{InternedScheduleLabel, LogLevel, ScheduleBuildSettings, ScheduleLabel},
+    prelude::*,
+    utils::HashMap,
+};
+use bevy_render::{pipelined_rendering::RenderExtractApp, Render, RenderApp};
+
+/// FIXME: bevy should not have any ambiguities, but it takes time to clean these up,
+/// so we're juste ignoring those for now.
+///
+/// See [#7386](https://github.com/bevyengine/bevy/issues/7386) for relevant issue.
+pub fn get_ignored_ambiguous_system_schedules() -> Vec<Box<dyn ScheduleLabel>> {
+    vec![
+        Box::new(First),
+        Box::new(PreUpdate),
+        //Box::new(Update),
+        Box::new(PostUpdate),
+        Box::new(Last),
+        Box::new(ExtractSchedule),
+        Box::new(Render),
+    ]
+}
+
+/// A test to confirm that `bevy` doesn't have system order ambiguity with [`DefaultPlugins`]
+/// This is run in CI to ensure that this doesn't regress again.
+pub fn main() {
+    let mut app = App::new();
+    app.add_plugins(DefaultPlugins);
+
+    let sub_app = app.main_mut();
+    configure_ambiguity_detection(sub_app);
+    let sub_app = app.sub_app_mut(RenderApp);
+    configure_ambiguity_detection(sub_app);
+    let sub_app = app.sub_app_mut(RenderExtractApp);
+    configure_ambiguity_detection(sub_app);
+
+    app.finish();
+    app.cleanup();
+    app.update();
+
+    let sub_app = app.main();
+
+    let ignored_schedules = get_ignored_ambiguous_system_schedules();
+
+    let ambiguities = count_ambiguities(sub_app);
+    let mut unexpected_ambiguities = vec![];
+    for kv in ambiguities.0.iter() {
+        if ignored_schedules.iter().any(|label| **label == **kv.0) {
+            continue;
+        }
+        if *kv.1 != 0 {
+            unexpected_ambiguities.push(kv.0);
+        }
+    }
```

Comment:
```suggestion
    for (&label, &count) in ambiguities.0.iter() {
        if ignored_schedules.contains(&label) {
            continue;
        }
        if count != 0 {
            unexpected_ambiguities.push(label);
        }
    }
```

Reply:
The `contains()` bit does not compile because we have `InternedScheduleLabel` on one side, and `Box<ScheduleLabel>` on the other ; I'm not sure how to convert from one to the other.